src/components/account/account.service.js

```diff
@@ -45,6 +45,45 @@ module.exports.login = async (credentials) => {
     }
 };
 
+/**
+ * Registers a new user account
+ * @param {Object} account Account data
+ * @returns {Object} User general data and jwt
+ * @throws {BadRequest} When the account data is not provided or the email already exists in the db
+ * @throws {InternalServerError} When there's an unexpected error.
+ */
+module.exports.register = async (account) => {
+    try {
+        if (!account)
+            throw new error.BadRequest('Account data not provided');
+
+        account.email = account.email.toLowerCase();
+
+        account.password = securityHelper.encrypt(account.password);
+
+        const createdAccount = await accountModel.create(account);
+
+        const tokenPayload = {
+            userId: createdAccount._id,
+            email: createdAccount.email,
+            timestamp: Date.now()
+        };
+
+        const jwt = securityHelper.generateJwt(tokenPayload);
+
+        return { access: jwt };
+    }
+    catch (err) {
+        if (err.name === 'MongoError' && err.code === 11000)
+            throw new error.BadRequest('Email already registered.');
+
+        if (!err.statusCode)
+            throw new error.InternalServerError('Unexpected error on login');
+
+        else throw err;
```

Comment:
Paula, your code is very good. I wouldnt change anything!

Just in case you forgot, i wanted to let you know that theres a possibility that if you throw the err variable itself, you could exposing a vulnerability.

I was taught that. If im wrong then forget about this!

Reply:
Good one! Luckily that will be taken care of by or error handling middleware, so we are safe (Y)